File: stack/0901_online_stock_span/0901_online_stock_span.go
package main

type StockSpanner struct {
	stack [][]int
}

func Constructor() StockSpanner {
	s := StockSpanner{}
	s.stack = make([][]int, 0)

	return s
}

func (this *StockSpanner) Next(price int) int {

	ans := 1
	for len(this.stack) > 0 && price >= this.stack[len(this.stack)-1][0] {
		ans = ans + this.stack[len(this.stack)-1][1]
		this.stack = this.stack[:len(this.stack)-1]
	}

	this.stack = append(this.stack, []int{price, ans})

	return ans
}

/**
 * Your StockSpanner object will be instantiated and called as such:
 * obj := Constructor();
 * param_1 := obj.Next(price);
 */

File: stack/0901_online_stock_span/0901_online_stock_span.py

class StockSpanner(object):
    def __init__(self):
        self.stack = []

    def next(self, price):
        weight = 1
        while self.stack and self.stack[-1][0] <= price:
            weight += self.stack.pop()[1]
        self.stack.append((price, weight))
        return weight